### PR TITLE
SLVSCODE-459 Check for ignored files in repository and its submodules

### PR DESCRIPTION
**Pull request by vojtech-suchy-sonarsource**
Running `git check-ignore` on a file in the repository's submodule completes with [128 error](https://git-scm.com/docs/git-check-ignore#Documentation/git-check-ignore.txt-128) saying _fatal: Pathspec XXX is in submodule_. 
To check if files are ignored in the repository and its submodules, we need

0) List of submodules (if non-present, do a regular `check-ignore` on all files)
1) Split files into two groups. Those in submodules (obtained in step 0) and those out of submodules
2) Run separate `check-ignore` for files in submodules and files outside of submodules
3) Merge the results from Step 2

**Known issues**
- If the submodule contains another submodule, the `check-ignore` returns [128 error]. For this case, we consider all files in submodules to be not ignored.

### CODE REVIEW

**Comment by kirill-knize-sonarsource** [comment]
Just checked on SonarJS repo that revealed the issue in the first place. Turned out they have nested submodules as well. But it's already an improvement and we filtering out good amount of files.

**Review by kirill-knize-sonarsource — APPROVED** [review]
Comments on logging and one minor. Else LGTM.

File: src/scm/scm.ts [hunk 1]
```diff
@@ -182,28 +185,91 @@ export async function filterOutScmIgnoredFiles(
...
         gitPath,
         gitArgs,
-        {cwd: workspaceFolderPath }
+        { cwd: workspaceFolderPath }
       );
       const onExit = (exitCode: number) => {
         if (exitCode === 1) {
           if (isVerboseEnabled()) {
-            logToSonarLintOutput(`Error on git command "${gitArgs}": ${stderr}`);
+            logToSonarLintOutput(`None of the provided paths are ignored.`);
```
[the remark this answers is not in the document]
Comment (vojtech-suchy-sonarsource) [reply src/scm/scm.ts]: Good point, I have removed the logging

File: src/scm/scm.ts [hunk 2]
```diff
@@ -182,28 +185,91 @@ export async function filterOutScmIgnoredFiles(
...
+
+    if (result.serr) {
+      return Promise.resolve([]);
+    }
+
+    const raw = result.sout;
+    return raw.split('\n').map(value => value.split(/\s/g)[1]).filter(value => value);
+  } catch (e) {
+    if (isVerboseEnabled()) {
+      logToSonarLintOutput(`No submodules found in '${repoPath}' repository.`);
```
[the remark this answers is not in the document]
Comment (vojtech-suchy-sonarsource) [reply src/scm/scm.ts]: Done

**Pull request merged**